herd/semExtra.ml

```diff
@@ -497,6 +499,34 @@ type concrete =
 
     type pp_barrier = { barrier:barrier ; pp:string; }
 
+  let patch_variant test variant cache_type =
```

Comment:
Could we consider renaming it something explicitly for cache-types variants? Like `patch_cache_type_variant`?

Reply:
Sounds right. Applying this comment.